File: readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    float letter  =  (letters / (float) words) *  100;
    float sentence = (sentences / (float) words) * 100;

    float index = 0.0588 * letter - 0.296 * sentence - 15.8;

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade 0.%f\n", round(index));
    }

}

int count_letters(string text)
{
    int letter_score = 0;

    // Compute score for each character
    for (int i = 0, mes = strlen(text); i < mes; i++)
    {
        if (isupper(text[i]))
        {
            letter_score++;
        }
        else if (islower(text[i]))
        {
            letter_score++;
        }
    }
    return letter_score;
}

int count_words(string text)
{
    int word_score = 1;

    for (int i = 0, mes = strlen(text);  i < mes; i++)
    {
        if (text[i] == ' ')
        {
            word_score++;
        }
    }
    // Return the number of words in text
    return  word_score;
}

int count_sentences(string text)
{

    int sentence_score  = 0;

    for (int i = 0, mes = strlen(text); i < mes; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_score++;
        }
    }
    // Return the number of sentences in text
    return  sentence_score;
}

File: readability.py


def main():

    text = input("Text: ")

    word = word_count(text)
    letter = letter_count(text)
    sentence = sentence_count(text)

    L = float(letter / word) * 100
    S = float(sentence / word) * 100

    formula = 0.0588 * L - 0.296 * S - 15.8

    if formula < 1:
        print("Before Grade 1")
    elif formula >= 16:
        print("Grade 16+")
    else:
        print("Grade %.0f" % round(formula))


def word_count(text):
    words = 1

    for i in range(len(text)):
        if text[i] == " ":
            words += 1

    return words


def letter_count(text):
    letters = 0

    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1

    return letters


def sentence_count(text):

    sentences = 0

    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()
